woocommerce/woocommerce
# reinstate `displayed` event for suggestions not displayed in metabox

- add func defining which suggestions contexts are hidden/revealable
- reinstate tracks event when rendering suggestion (for non-hidden
suggestions)
- add check in (context-sensitive) tab click handler to abort if
that context has already fired displayed events

### All Submissions:


### Changes proposed in this Pull Request:
Adds back tracks event when rendering events. This is now conditional on whether the suggestion context is visible on page load. 

https://github.com/woocommerce/woocommerce/pull/23253 moved the tracks event into a click handler, to ensure that it's only fired when product edit metabox suggestions are actually visible. 

This PR adds back the original code when it's needed – i.e. for products and orders empty states.

### How to test the changes in this Pull Request:
1. Get this branch up & running as your WooCommerce plugin.
2. View products empty state – note the tracks events (e.g. in Network tab) and check their attributes. See example below.
3. View orders empty state - check tracks events are correct.
4. View product edit page - note there should still be no `marketplace_suggestion_displayed` on page load.
4. Scroll down & click `Get more options` tab in metabox. Check tracks events are displayed and attributes are correct.
3. Click and interact with various suggestions (click link/button, dismiss) & check that appropriate events are fired, with correct attributes.

<img width="722" alt="Screen Shot 2019-04-10 at 9 48 48 AM" src="https://user-images.githubusercontent.com/4167300/55837570-e0a93e80-5b75-11e9-9d1a-30883e92b5de.png">
